cub/cub/agent/agent_reduce_by_key.cuh

```diff
@@ -551,7 +551,7 @@ struct AgentReduceByKey
       //   Subsequent tiles get last key from previous tile
       if constexpr (is_streaming_invocation)
       {
-        tile_predecessor = (tile_idx == 0) ? streaming_context.predecessor_key() : d_keys_in[tile_offset - 1];
+        tile_predecessor = (tile_idx == 0) ? streaming_context.value().predecessor_key() : d_keys_in[tile_offset - 1];
```

Comment:
Remark: `.value()` performs a check inside. If this is leading to SASS changs, consider just reaching into the optional using `->`, like:
```suggestion
        tile_predecessor = (tile_idx == 0) ? streaming_context->predecessor_key() : d_keys_in[tile_offset - 1];
```
This only applies, if we safely know that the optional contains a value (but I think we know via `if constexpr (is_streaming_invocation)`)

Reply:
That said, it may still need to be considered depending on the benchmark outcome.